geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala

```diff
@@ -334,6 +335,22 @@ class HBaseDataStoreTest extends HBaseTest with LazyLogging {
         SchemaProperties.CheckDistributedVersion.clear()
       }
     }
+
+    "support configuration via path or embedded xml" in {
+      val embedded =
+        """<configuration>
+          |  <property>
+          |    <name>foo.embedded</name>
+          |    <value>bar</value>
+          |  </property>
+          |</configuration>""".stripMargin
+      val resource = new File(getClass.getClassLoader.getResource("custom-site.xml").toURI).getAbsolutePath
+      val params = Map(ConfigsParam.getName -> embedded, ConfigPathsParam.getName -> resource)
+      val conf = HBaseConnectionPool.getConfiguration(params.asJava)
+
+      conf.get("foo.embedded") mustEqual "bar"
+      conf.get("foo.resource") mustEqual "baz"
```

Comment:
How does this interact with any *-site.xml files which may be in the environment?

Reply:
it will load the default resources as per usual. I *think* these additional ones will take precedence if there is a conflict, since they're loaded second, but I'm not sure about that and wouldn't guarantee it.